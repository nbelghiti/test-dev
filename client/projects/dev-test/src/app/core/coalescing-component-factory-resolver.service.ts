/* tslint:disable */
/**
 * See https://github.com/jonrimmer/angular-coalescing-component-factory-resolver/blob/master/src/app/coalescing-component-factory-resolver.service.ts
 * Discussion & bug ticket: https://github.com/angular/angular/issues/14324#issuecomment-481898762
 *
 * Entry components are only available within a lazy loaded module, breaking dynamic generation (e.g. dialogs). This monkey patch enables resolving entry components from lazy loaded modules as well.
 *
 *
 * Provide in root module:
 *
 * @NgModule({
 *   providers: [CoalescingComponentFactoryResolver]
 * })
 * class AppModule {
 *   constructor(coalescingResolver: CoalescingComponentFactoryResolver) {
 *     coalescingResolver.init();
 *   }
 * }
 *
 *
 * Use in lazy loaded module:
 *
 *
 * @NgModule({})
 * export class LazyModule {
 *   constructor(
 *     coalescingResolver: CoalescingComponentFactoryResolver,
 *     localResolver: ComponentFactoryResolver
 *   ) {
 *     coalescingResolver.registerResolver(localResolver);
 *   }
 * }
 *
 *
 * How it works: When initialised, it injects the root app's ComponentFactoryResolver and monkey patches the resolveComponentFactory method
 * to call its own implementation. This implementation first tries resolving the component factory on all the registered lazy-module resolvers,
 * then falls back to the root app resolver (there's a bit of extra logic to avoid cyclic calls, but that's the gist).
 */
/* tslint:enable */

import {
	ComponentFactory,
	ComponentFactoryResolver,
	Injectable,
	Type
} from '@angular/core';

@Injectable()
export class CoalescingComponentFactoryResolver extends ComponentFactoryResolver {
	private rootResolve: (component: Type<any>) => ComponentFactory<any>;

	private inCall = false;

	private readonly resolvers = new Map<
		ComponentFactoryResolver,
		(component: Type<any>) => ComponentFactory<any>
	>();

	constructor(private readonly rootResolver: ComponentFactoryResolver) {
		super();
	}

	init() {
		this.rootResolve = this.rootResolver.resolveComponentFactory;
		this.rootResolver.resolveComponentFactory = this.resolveComponentFactory;
	}

	public registerResolver(resolver: ComponentFactoryResolver) {
		const original = resolver.resolveComponentFactory;
		this.resolvers.set(resolver, original);
	}

	public resolveComponentFactory = <T>(
		component: Type<T>
	): ComponentFactory<T> => {
		// Prevents cyclic calls.
		if (this.inCall) {
			return null;
		}

		this.inCall = true;
		try {
			const result = this.resolveInternal(component);
			return result;
		} finally {
			this.inCall = false;
		}
	}

	private resolveInternal = <T>(component: Type<T>): ComponentFactory<T> => {
		for (const [resolver, fn] of Array.from(this.resolvers.entries())) {
			try {
				const factory = fn.call(resolver, component);
				if (factory) {
					return factory;
				}
			} catch { }
		}

		return this.rootResolve.call(this.rootResolver, component);
	}
}
