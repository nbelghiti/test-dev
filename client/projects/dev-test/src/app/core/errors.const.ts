export const ToastrConfig = {
	maxOpened: 2,
	timeOut: 8000
};
