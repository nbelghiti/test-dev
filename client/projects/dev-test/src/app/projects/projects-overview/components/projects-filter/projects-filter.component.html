<form class="l-form" [formGroup]="selectFilterForm">
	<div class="l-form__grid">
		<div class="l-grid l-grid--with-spacing l-grid--stretch">
			<div class="l-grid__col--3">
				<ui-filter>
					<ui-filter-toggle
						[label]="'MY_PROJECTS__FILTER__PROJECT_PHASE' | translate"
						background="grey"
						[currentOption]="statusKeyValue ? statusKeyValue?.label : '- All'"
					>
					</ui-filter-toggle>

					<ui-filter-menu
						[items]="projectStatusOptions"
						formControlName="status"
					></ui-filter-menu>
				</ui-filter>
			</div>

			<div class="l-grid__col--3">
				<ui-filter>
					<ui-filter-toggle
						[label]="'MY_PROJECTS__FILTER__LOCATION' | translate"
						background="grey"
						[currentOption]="locationKeyValue ? locationKeyValue?.label : '- All'"
					>
					</ui-filter-toggle>

					<ui-filter-menu
						[items]="projectLocationOptions"
						formControlName="location"
					></ui-filter-menu>
				</ui-filter>
			</div>
		</div>
	</div>
</form>
