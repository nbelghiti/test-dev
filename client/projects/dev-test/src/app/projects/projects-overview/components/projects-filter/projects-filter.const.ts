export const ProjectStatusOptions = [{
	label: '- All',
	value: '- All',
}, {
	label: 'Bidding',
	value: 'bidding',
}, {
	label: 'Engineering',
	value: 'engineering',
}, {
	label: 'Construction',
	value: 'construction',
}, {
	label: 'Pac',
	value: 'pac',
}, {
	label: 'Fac',
	value: 'fac',
}];

export const ProjectLocationOptions = [{
	label: '- All',
	value: '- All',
}, {
	label: 'Bahrain',
	value: 'Bahrain',
}, {
	label: 'Canada',
	value: 'Canada',
}, {
	label: 'France',
	value: 'France',
}, {
	label: 'Italy',
	value: 'Italy',
}, {
	label: 'Saudi Arabia',
	value: 'Saudi Arabia',
}, {
	label: 'Taiwan',
	value: 'Taiwan',
}, {
	label: 'USA, Nevada',
	value: 'USA, Nevada',
}, {
	label: 'USA, Oklahoma',
	value: 'USA, Oklahoma',
}, {
	label: 'USA, Pennsylvania',
	value: 'USA, Pennsylvania',
}];
