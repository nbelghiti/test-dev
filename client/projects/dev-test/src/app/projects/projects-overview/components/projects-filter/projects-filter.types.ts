export interface IProjectFilterOption {
	label: string;
	value: string;
}

export interface IProjectFilters {
	status: IProjectFilterOption;
	location: IProjectFilterOption;
}
