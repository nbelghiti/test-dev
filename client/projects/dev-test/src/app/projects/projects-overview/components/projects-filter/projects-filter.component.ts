import { Component, OnDestroy, OnInit, forwardRef } from '@angular/core';
import { ControlValueAccessor, FormControl, FormGroup, NG_VALUE_ACCESSOR } from '@angular/forms';
import { isNil, propOr } from 'ramda';
import { Subject } from 'rxjs';
import { takeUntil } from 'rxjs/operators';

import { IProjectsFilter } from '~/repositories/projects/services/projects.types';

import { ProjectLocationOptions, ProjectStatusOptions } from './projects-filter.const';
import { IProjectFilterOption, IProjectFilters } from './projects-filter.types';

@Component({
	selector: 'adb-projects-filter',
	templateUrl: './projects-filter.component.html',
	providers: [
		{
			provide: NG_VALUE_ACCESSOR,
			useExisting: forwardRef(() => ProjectsFilterComponent),
			multi: true,
		},
	],
})
export class ProjectsFilterComponent implements ControlValueAccessor, OnInit, OnDestroy {
	public projectStatusOptions = ProjectStatusOptions;
	public projectLocationOptions = ProjectLocationOptions;
	public value: IProjectsFilter;
	public selectFilterForm: FormGroup;
	public statusKeyValue: IProjectFilterOption;
	public locationKeyValue: IProjectFilterOption;

	private onDestroy$: Subject<boolean> = new Subject<boolean>();

	public ngOnInit(): void {
		this.selectFilterForm = new FormGroup({
			status: new FormControl(null),
			location: new FormControl(null),
		});

		this.selectFilterForm.valueChanges
			.pipe(
				takeUntil(this.onDestroy$),
			)
			.subscribe((formData: IProjectFilters) => {
				Object.keys(formData).forEach((filter: any) => {
					formData[filter] = propOr(null, ['value'], formData[filter]) === '- All' ? null : propOr(null, ['value'], formData[filter]);
				});

				this.onChange(formData);
			});
	}

	public ngOnDestroy(): void {
		this.onDestroy$.next(true);
		this.onDestroy$.complete();
	}

	public onChange: (_) => any = () => { };

	public onTouched = () => { };

	public writeValue(value: IProjectsFilter): void {
		this.statusKeyValue = this.findKeyValue(this.projectStatusOptions, value, 'status');

		this.locationKeyValue = this.findKeyValue(this.projectLocationOptions, value, 'location');

		this.selectFilterForm.patchValue({
			status: this.statusKeyValue,
			location: this.locationKeyValue,
		}, { emitEvent: false, onlySelf: true });
	}

	public registerOnChange(onChange: (_) => any): void {
		this.onChange = onChange;
	}

	public registerOnTouched(fn: () => void): void {
		this.onTouched = fn;
	}

	private findKeyValue(options: IProjectFilterOption[], value: IProjectsFilter, property: string): IProjectFilterOption {
		const matchingOption = options.find((option: IProjectFilterOption) => option.value === propOr(null, property, value));

		if (propOr('', ['value'], matchingOption) === '- All' || isNil(propOr('', ['value'], matchingOption))) {
			return null;
		}

		return matchingOption;
	}
}
