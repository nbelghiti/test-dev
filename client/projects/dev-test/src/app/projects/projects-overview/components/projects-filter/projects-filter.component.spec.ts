import { HttpClientTestingModule } from '@angular/common/http/testing';
import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { TranslateModule } from '@ngx-translate/core';

import { FilterModule } from 'adb-ui';

import { ProjectsFilterComponent } from './projects-filter.component';

describe('ProjectsFilterComponent', () => {
	let component: ProjectsFilterComponent;
	let fixture: ComponentFixture<ProjectsFilterComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				// Begin: UI Modules
				FilterModule,
				// End: UI Modules
				HttpClientTestingModule,
				TranslateModule.forRoot(),
				FormsModule,
				ReactiveFormsModule,
			],
			declarations: [ ProjectsFilterComponent ]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(ProjectsFilterComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
