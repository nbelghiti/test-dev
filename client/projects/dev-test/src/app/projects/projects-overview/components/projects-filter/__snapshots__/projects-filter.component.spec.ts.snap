// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsFilterComponent Should render correctly 1`] = `
<adb-projects-filter
  onChange={[Function Function]}
  onDestroy$={[Function Subject]}
  onTouched={[Function Function]}
  projectLocationOptions={[Function Array]}
  projectStatusOptions={[Function Array]}
  selectFilterForm={[Function FormGroup]}
>
  <form
    class="l-form ng-untouched ng-pristine ng-valid"
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    <div
      class="l-form__grid"
    >
      <div
        class="l-grid l-grid--with-spacing l-grid--stretch"
      >
        <div
          class="l-grid__col--3"
        >
          <ui-filter>
            <div
              class="c-filter c-flyout is-closed"
              uiflyout=""
            >
              <div
                class="c-flyout__toggle"
                uiflyouttoggle=""
              >
                <ui-filter-toggle
                  _nghost-a-c4=""
                  background="grey"
                  ng-reflect-background="grey"
                  ng-reflect-current-option="- All"
                  ng-reflect-label="MY_PROJECTS__FILTER__PROJECT_P"
                >
                  <div
                    _ngcontent-a-c4=""
                    class="c-filter-toggle c-filter-toggle--grey"
                    ng-reflect-klass="c-filter-toggle"
                    ng-reflect-ng-class="[object Object]"
                  >
                    
                    <span
                      _ngcontent-a-c4=""
                      class="c-filter-toggle__label"
                    >
                      MY_PROJECTS__FILTER__PROJECT_PHASE
                    </span>
                    <button
                      _ngcontent-a-c4=""
                      class="c-filter-toggle__trigger"
                      type="button"
                    >
                      <span
                        _ngcontent-a-c4=""
                        class="c-filter-toggle__current-option"
                      >
                        - All
                      </span>
                      <div
                        _ngcontent-a-c4=""
                        class="c-filter-toggle__icon"
                      >
                        <ui-icon
                          _ngcontent-a-c4=""
                          _nghost-a-c6=""
                          class="c-icon"
                          name="caret-down"
                          ng-reflect-name="caret-down"
                          ng-reflect-size="small"
                          size="small"
                        >
                          <i
                            _ngcontent-a-c6=""
                            class="c-icon__glyph c-icon__glyph--small  fas fa-caret-down"
                          />
                        </ui-icon>
                      </div>
                    </button>
                  </div>
                </ui-filter-toggle>
              </div>
              <div
                class="c-filter__content c-flyout__content--left c-flyout__content c-flyout__toggle"
                ng-reflect-class="c-filter__content c-flyout__co"
                ng-reflect-toggle-state="CLOSED"
                uiflyoutcontent=""
                uiflyouttoggle=""
              >
                <ui-filter-menu
                  _nghost-a-c5=""
                  class="ng-untouched ng-pristine ng-valid"
                  formcontrolname="status"
                  ng-reflect-items="[object Object],[object Object"
                  ng-reflect-name="status"
                >
                  <ul
                    _ngcontent-a-c5=""
                    class="c-filter-menu"
                  >
                    
                    <li
                      _ngcontent-a-c5=""
                      class="c-filter-menu__item"
                    >
                      <ui-filter-menu-item
                        _ngcontent-a-c5=""
                        _nghost-a-c7=""
                        ng-reflect-active="true"
                        ng-reflect-label="- All"
                      >
                        
                        <button
                          _ngcontent-a-c7=""
                          class="c-filter-menu-item c-filter-menu-item--active"
                          ng-reflect-klass="c-filter-menu-item"
                          ng-reflect-ng-class="[object Object]"
                          title="- All"
                        >
                          <span
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__value"
                          >
                            - All
                          </span>
                          <div
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__icon"
                          >
                            <ui-icon
                              _ngcontent-a-c7=""
                              _nghost-a-c6=""
                              class="c-icon"
                              name="check"
                              ng-reflect-name="check"
                            >
                              <i
                                _ngcontent-a-c6=""
                                class="c-icon__glyph   fas fa-check"
                              />
                            </ui-icon>
                          </div>
                        </button>
                        
                      </ui-filter-menu-item>
                    </li>
                    <li
                      _ngcontent-a-c5=""
                      class="c-filter-menu__item"
                    >
                      <ui-filter-menu-item
                        _ngcontent-a-c5=""
                        _nghost-a-c7=""
                        ng-reflect-active="false"
                        ng-reflect-label="Bidding"
                      >
                        
                        <button
                          _ngcontent-a-c7=""
                          class="c-filter-menu-item"
                          ng-reflect-klass="c-filter-menu-item"
                          ng-reflect-ng-class="[object Object]"
                          title="Bidding"
                        >
                          <span
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__value"
                          >
                            Bidding
                          </span>
                          <div
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__icon"
                          >
                            <ui-icon
                              _ngcontent-a-c7=""
                              _nghost-a-c6=""
                              class="c-icon"
                              name="check"
                              ng-reflect-name="check"
                            >
                              <i
                                _ngcontent-a-c6=""
                                class="c-icon__glyph   fas fa-check"
                              />
                            </ui-icon>
                          </div>
                        </button>
                        
                      </ui-filter-menu-item>
                    </li>
                    <li
                      _ngcontent-a-c5=""
                      class="c-filter-menu__item"
                    >
                      <ui-filter-menu-item
                        _ngcontent-a-c5=""
                        _nghost-a-c7=""
                        ng-reflect-active="false"
                        ng-reflect-label="Engineering"
                      >
                        
                        <button
                          _ngcontent-a-c7=""
                          class="c-filter-menu-item"
                          ng-reflect-klass="c-filter-menu-item"
                          ng-reflect-ng-class="[object Object]"
                          title="Engineering"
                        >
                          <span
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__value"
                          >
                            Engineering
                          </span>
                          <div
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__icon"
                          >
                            <ui-icon
                              _ngcontent-a-c7=""
                              _nghost-a-c6=""
                              class="c-icon"
                              name="check"
                              ng-reflect-name="check"
                            >
                              <i
                                _ngcontent-a-c6=""
                                class="c-icon__glyph   fas fa-check"
                              />
                            </ui-icon>
                          </div>
                        </button>
                        
                      </ui-filter-menu-item>
                    </li>
                    <li
                      _ngcontent-a-c5=""
                      class="c-filter-menu__item"
                    >
                      <ui-filter-menu-item
                        _ngcontent-a-c5=""
                        _nghost-a-c7=""
                        ng-reflect-active="false"
                        ng-reflect-label="Construction"
                      >
                        
                        <button
                          _ngcontent-a-c7=""
                          class="c-filter-menu-item"
                          ng-reflect-klass="c-filter-menu-item"
                          ng-reflect-ng-class="[object Object]"
                          title="Construction"
                        >
                          <span
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__value"
                          >
                            Construction
                          </span>
                          <div
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__icon"
                          >
                            <ui-icon
                              _ngcontent-a-c7=""
                              _nghost-a-c6=""
                              class="c-icon"
                              name="check"
                              ng-reflect-name="check"
                            >
                              <i
                                _ngcontent-a-c6=""
                                class="c-icon__glyph   fas fa-check"
                              />
                            </ui-icon>
                          </div>
                        </button>
                        
                      </ui-filter-menu-item>
                    </li>
                    <li
                      _ngcontent-a-c5=""
                      class="c-filter-menu__item"
                    >
                      <ui-filter-menu-item
                        _ngcontent-a-c5=""
                        _nghost-a-c7=""
                        ng-reflect-active="false"
                        ng-reflect-label="Pac"
                      >
                        
                        <button
                          _ngcontent-a-c7=""
                          class="c-filter-menu-item"
                          ng-reflect-klass="c-filter-menu-item"
                          ng-reflect-ng-class="[object Object]"
                          title="Pac"
                        >
                          <span
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__value"
                          >
                            Pac
                          </span>
                          <div
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__icon"
                          >
                            <ui-icon
                              _ngcontent-a-c7=""
                              _nghost-a-c6=""
                              class="c-icon"
                              name="check"
                              ng-reflect-name="check"
                            >
                              <i
                                _ngcontent-a-c6=""
                                class="c-icon__glyph   fas fa-check"
                              />
                            </ui-icon>
                          </div>
                        </button>
                        
                      </ui-filter-menu-item>
                    </li>
                    <li
                      _ngcontent-a-c5=""
                      class="c-filter-menu__item"
                    >
                      <ui-filter-menu-item
                        _ngcontent-a-c5=""
                        _nghost-a-c7=""
                        ng-reflect-active="false"
                        ng-reflect-label="Fac"
                      >
                        
                        <button
                          _ngcontent-a-c7=""
                          class="c-filter-menu-item"
                          ng-reflect-klass="c-filter-menu-item"
                          ng-reflect-ng-class="[object Object]"
                          title="Fac"
                        >
                          <span
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__value"
                          >
                            Fac
                          </span>
                          <div
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__icon"
                          >
                            <ui-icon
                              _ngcontent-a-c7=""
                              _nghost-a-c6=""
                              class="c-icon"
                              name="check"
                              ng-reflect-name="check"
                            >
                              <i
                                _ngcontent-a-c6=""
                                class="c-icon__glyph   fas fa-check"
                              />
                            </ui-icon>
                          </div>
                        </button>
                        
                      </ui-filter-menu-item>
                    </li>
                  </ul>
                </ui-filter-menu>
              </div>
            </div>
          </ui-filter>
        </div>
        <div
          class="l-grid__col--3"
        >
          <ui-filter>
            <div
              class="c-filter c-flyout is-closed"
              uiflyout=""
            >
              <div
                class="c-flyout__toggle"
                uiflyouttoggle=""
              >
                <ui-filter-toggle
                  _nghost-a-c4=""
                  background="grey"
                  ng-reflect-background="grey"
                  ng-reflect-current-option="- All"
                  ng-reflect-label="MY_PROJECTS__FILTER__LOCATION"
                >
                  <div
                    _ngcontent-a-c4=""
                    class="c-filter-toggle c-filter-toggle--grey"
                    ng-reflect-klass="c-filter-toggle"
                    ng-reflect-ng-class="[object Object]"
                  >
                    
                    <span
                      _ngcontent-a-c4=""
                      class="c-filter-toggle__label"
                    >
                      MY_PROJECTS__FILTER__LOCATION
                    </span>
                    <button
                      _ngcontent-a-c4=""
                      class="c-filter-toggle__trigger"
                      type="button"
                    >
                      <span
                        _ngcontent-a-c4=""
                        class="c-filter-toggle__current-option"
                      >
                        - All
                      </span>
                      <div
                        _ngcontent-a-c4=""
                        class="c-filter-toggle__icon"
                      >
                        <ui-icon
                          _ngcontent-a-c4=""
                          _nghost-a-c6=""
                          class="c-icon"
                          name="caret-down"
                          ng-reflect-name="caret-down"
                          ng-reflect-size="small"
                          size="small"
                        >
                          <i
                            _ngcontent-a-c6=""
                            class="c-icon__glyph c-icon__glyph--small  fas fa-caret-down"
                          />
                        </ui-icon>
                      </div>
                    </button>
                  </div>
                </ui-filter-toggle>
              </div>
              <div
                class="c-filter__content c-flyout__content--left c-flyout__content c-flyout__toggle"
                ng-reflect-class="c-filter__content c-flyout__co"
                ng-reflect-toggle-state="CLOSED"
                uiflyoutcontent=""
                uiflyouttoggle=""
              >
                <ui-filter-menu
                  _nghost-a-c5=""
                  class="ng-untouched ng-pristine ng-valid"
                  formcontrolname="location"
                  ng-reflect-items="[object Object],[object Object"
                  ng-reflect-name="location"
                >
                  <ul
                    _ngcontent-a-c5=""
                    class="c-filter-menu"
                  >
                    
                    <li
                      _ngcontent-a-c5=""
                      class="c-filter-menu__item"
                    >
                      <ui-filter-menu-item
                        _ngcontent-a-c5=""
                        _nghost-a-c7=""
                        ng-reflect-active="true"
                        ng-reflect-label="- All"
                      >
                        
                        <button
                          _ngcontent-a-c7=""
                          class="c-filter-menu-item c-filter-menu-item--active"
                          ng-reflect-klass="c-filter-menu-item"
                          ng-reflect-ng-class="[object Object]"
                          title="- All"
                        >
                          <span
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__value"
                          >
                            - All
                          </span>
                          <div
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__icon"
                          >
                            <ui-icon
                              _ngcontent-a-c7=""
                              _nghost-a-c6=""
                              class="c-icon"
                              name="check"
                              ng-reflect-name="check"
                            >
                              <i
                                _ngcontent-a-c6=""
                                class="c-icon__glyph   fas fa-check"
                              />
                            </ui-icon>
                          </div>
                        </button>
                        
                      </ui-filter-menu-item>
                    </li>
                    <li
                      _ngcontent-a-c5=""
                      class="c-filter-menu__item"
                    >
                      <ui-filter-menu-item
                        _ngcontent-a-c5=""
                        _nghost-a-c7=""
                        ng-reflect-active="false"
                        ng-reflect-label="Bahrain"
                      >
                        
                        <button
                          _ngcontent-a-c7=""
                          class="c-filter-menu-item"
                          ng-reflect-klass="c-filter-menu-item"
                          ng-reflect-ng-class="[object Object]"
                          title="Bahrain"
                        >
                          <span
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__value"
                          >
                            Bahrain
                          </span>
                          <div
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__icon"
                          >
                            <ui-icon
                              _ngcontent-a-c7=""
                              _nghost-a-c6=""
                              class="c-icon"
                              name="check"
                              ng-reflect-name="check"
                            >
                              <i
                                _ngcontent-a-c6=""
                                class="c-icon__glyph   fas fa-check"
                              />
                            </ui-icon>
                          </div>
                        </button>
                        
                      </ui-filter-menu-item>
                    </li>
                    <li
                      _ngcontent-a-c5=""
                      class="c-filter-menu__item"
                    >
                      <ui-filter-menu-item
                        _ngcontent-a-c5=""
                        _nghost-a-c7=""
                        ng-reflect-active="false"
                        ng-reflect-label="Canada"
                      >
                        
                        <button
                          _ngcontent-a-c7=""
                          class="c-filter-menu-item"
                          ng-reflect-klass="c-filter-menu-item"
                          ng-reflect-ng-class="[object Object]"
                          title="Canada"
                        >
                          <span
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__value"
                          >
                            Canada
                          </span>
                          <div
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__icon"
                          >
                            <ui-icon
                              _ngcontent-a-c7=""
                              _nghost-a-c6=""
                              class="c-icon"
                              name="check"
                              ng-reflect-name="check"
                            >
                              <i
                                _ngcontent-a-c6=""
                                class="c-icon__glyph   fas fa-check"
                              />
                            </ui-icon>
                          </div>
                        </button>
                        
                      </ui-filter-menu-item>
                    </li>
                    <li
                      _ngcontent-a-c5=""
                      class="c-filter-menu__item"
                    >
                      <ui-filter-menu-item
                        _ngcontent-a-c5=""
                        _nghost-a-c7=""
                        ng-reflect-active="false"
                        ng-reflect-label="France"
                      >
                        
                        <button
                          _ngcontent-a-c7=""
                          class="c-filter-menu-item"
                          ng-reflect-klass="c-filter-menu-item"
                          ng-reflect-ng-class="[object Object]"
                          title="France"
                        >
                          <span
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__value"
                          >
                            France
                          </span>
                          <div
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__icon"
                          >
                            <ui-icon
                              _ngcontent-a-c7=""
                              _nghost-a-c6=""
                              class="c-icon"
                              name="check"
                              ng-reflect-name="check"
                            >
                              <i
                                _ngcontent-a-c6=""
                                class="c-icon__glyph   fas fa-check"
                              />
                            </ui-icon>
                          </div>
                        </button>
                        
                      </ui-filter-menu-item>
                    </li>
                    <li
                      _ngcontent-a-c5=""
                      class="c-filter-menu__item"
                    >
                      <ui-filter-menu-item
                        _ngcontent-a-c5=""
                        _nghost-a-c7=""
                        ng-reflect-active="false"
                        ng-reflect-label="Italy"
                      >
                        
                        <button
                          _ngcontent-a-c7=""
                          class="c-filter-menu-item"
                          ng-reflect-klass="c-filter-menu-item"
                          ng-reflect-ng-class="[object Object]"
                          title="Italy"
                        >
                          <span
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__value"
                          >
                            Italy
                          </span>
                          <div
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__icon"
                          >
                            <ui-icon
                              _ngcontent-a-c7=""
                              _nghost-a-c6=""
                              class="c-icon"
                              name="check"
                              ng-reflect-name="check"
                            >
                              <i
                                _ngcontent-a-c6=""
                                class="c-icon__glyph   fas fa-check"
                              />
                            </ui-icon>
                          </div>
                        </button>
                        
                      </ui-filter-menu-item>
                    </li>
                    <li
                      _ngcontent-a-c5=""
                      class="c-filter-menu__item"
                    >
                      <ui-filter-menu-item
                        _ngcontent-a-c5=""
                        _nghost-a-c7=""
                        ng-reflect-active="false"
                        ng-reflect-label="Saudi Arabia"
                      >
                        
                        <button
                          _ngcontent-a-c7=""
                          class="c-filter-menu-item"
                          ng-reflect-klass="c-filter-menu-item"
                          ng-reflect-ng-class="[object Object]"
                          title="Saudi Arabia"
                        >
                          <span
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__value"
                          >
                            Saudi Arabia
                          </span>
                          <div
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__icon"
                          >
                            <ui-icon
                              _ngcontent-a-c7=""
                              _nghost-a-c6=""
                              class="c-icon"
                              name="check"
                              ng-reflect-name="check"
                            >
                              <i
                                _ngcontent-a-c6=""
                                class="c-icon__glyph   fas fa-check"
                              />
                            </ui-icon>
                          </div>
                        </button>
                        
                      </ui-filter-menu-item>
                    </li>
                    <li
                      _ngcontent-a-c5=""
                      class="c-filter-menu__item"
                    >
                      <ui-filter-menu-item
                        _ngcontent-a-c5=""
                        _nghost-a-c7=""
                        ng-reflect-active="false"
                        ng-reflect-label="Taiwan"
                      >
                        
                        <button
                          _ngcontent-a-c7=""
                          class="c-filter-menu-item"
                          ng-reflect-klass="c-filter-menu-item"
                          ng-reflect-ng-class="[object Object]"
                          title="Taiwan"
                        >
                          <span
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__value"
                          >
                            Taiwan
                          </span>
                          <div
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__icon"
                          >
                            <ui-icon
                              _ngcontent-a-c7=""
                              _nghost-a-c6=""
                              class="c-icon"
                              name="check"
                              ng-reflect-name="check"
                            >
                              <i
                                _ngcontent-a-c6=""
                                class="c-icon__glyph   fas fa-check"
                              />
                            </ui-icon>
                          </div>
                        </button>
                        
                      </ui-filter-menu-item>
                    </li>
                    <li
                      _ngcontent-a-c5=""
                      class="c-filter-menu__item"
                    >
                      <ui-filter-menu-item
                        _ngcontent-a-c5=""
                        _nghost-a-c7=""
                        ng-reflect-active="false"
                        ng-reflect-label="USA, Nevada"
                      >
                        
                        <button
                          _ngcontent-a-c7=""
                          class="c-filter-menu-item"
                          ng-reflect-klass="c-filter-menu-item"
                          ng-reflect-ng-class="[object Object]"
                          title="USA, Nevada"
                        >
                          <span
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__value"
                          >
                            USA, Nevada
                          </span>
                          <div
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__icon"
                          >
                            <ui-icon
                              _ngcontent-a-c7=""
                              _nghost-a-c6=""
                              class="c-icon"
                              name="check"
                              ng-reflect-name="check"
                            >
                              <i
                                _ngcontent-a-c6=""
                                class="c-icon__glyph   fas fa-check"
                              />
                            </ui-icon>
                          </div>
                        </button>
                        
                      </ui-filter-menu-item>
                    </li>
                    <li
                      _ngcontent-a-c5=""
                      class="c-filter-menu__item"
                    >
                      <ui-filter-menu-item
                        _ngcontent-a-c5=""
                        _nghost-a-c7=""
                        ng-reflect-active="false"
                        ng-reflect-label="USA, Oklahoma"
                      >
                        
                        <button
                          _ngcontent-a-c7=""
                          class="c-filter-menu-item"
                          ng-reflect-klass="c-filter-menu-item"
                          ng-reflect-ng-class="[object Object]"
                          title="USA, Oklahoma"
                        >
                          <span
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__value"
                          >
                            USA, Oklahoma
                          </span>
                          <div
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__icon"
                          >
                            <ui-icon
                              _ngcontent-a-c7=""
                              _nghost-a-c6=""
                              class="c-icon"
                              name="check"
                              ng-reflect-name="check"
                            >
                              <i
                                _ngcontent-a-c6=""
                                class="c-icon__glyph   fas fa-check"
                              />
                            </ui-icon>
                          </div>
                        </button>
                        
                      </ui-filter-menu-item>
                    </li>
                    <li
                      _ngcontent-a-c5=""
                      class="c-filter-menu__item"
                    >
                      <ui-filter-menu-item
                        _ngcontent-a-c5=""
                        _nghost-a-c7=""
                        ng-reflect-active="false"
                        ng-reflect-label="USA, Pennsylvania"
                      >
                        
                        <button
                          _ngcontent-a-c7=""
                          class="c-filter-menu-item"
                          ng-reflect-klass="c-filter-menu-item"
                          ng-reflect-ng-class="[object Object]"
                          title="USA, Pennsylvania"
                        >
                          <span
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__value"
                          >
                            USA, Pennsylvania
                          </span>
                          <div
                            _ngcontent-a-c7=""
                            class="c-filter-menu-item__icon"
                          >
                            <ui-icon
                              _ngcontent-a-c7=""
                              _nghost-a-c6=""
                              class="c-icon"
                              name="check"
                              ng-reflect-name="check"
                            >
                              <i
                                _ngcontent-a-c6=""
                                class="c-icon__glyph   fas fa-check"
                              />
                            </ui-icon>
                          </div>
                        </button>
                        
                      </ui-filter-menu-item>
                    </li>
                  </ul>
                </ui-filter-menu>
              </div>
            </div>
          </ui-filter>
        </div>
      </div>
    </div>
  </form>
</adb-projects-filter>
`;
